.github/workflows/codeql.yml

```diff
@@ -0,0 +1,51 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "master" ]
+  workflow_dispatch:
+  schedule:
+    - cron: '15 14 * * 1'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-20.04
+    container: seladb/ubuntu2204
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp' ]
```

Comment:
is `matrix` needed here? there's only one language...

Reply:
Makes it easier to add additional languages if one day you would like to scan python too for example.

Reply:
I don't think we'll need to examine the python code because it's mostly for tests or configuration, but we can keep it as is